packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -234,62 +292,29 @@ function* doStartPurchaseContentFlow({
     const extraAmountBN = new BN(extraAmount ?? 0).mul(BN_USDC_CENT_WEI)
     const totalAmountDueCentsBN = priceBN.add(extraAmountBN) as BNUSDC
 
-    const balanceNeeded = getBalanceNeeded(
-      totalAmountDueCentsBN,
-      new BN(initialBalance.toString()) as BNUSDC,
-      usdcConfig.minUSDCPurchaseAmountCents
-    )
-
-    // buy USDC if necessary
-    if (balanceNeeded.gtn(0)) {
-      const balanceNeededCents = ceilingBNUSDCToNearestCent(balanceNeeded)
-        .div(BN_USDC_CENT_WEI)
-        .toNumber()
+    if (purchaseMethod === PurchaseMethod.CARD) {
+      // Transition to 'buying USDC' stage
       yield* put(buyUSDC())
-      let result: RaceStatusResult | null = null
-      if (isBuyUSDCViaSolEnabled) {
-        yield* put(
-          buyCryptoViaSol({
-            // expects "friendly" amount, so dollars
-            amount: balanceNeededCents / 100.0,
-            mint: 'usdc',
-            provider: OnRampProvider.STRIPE
-          })
-        )
-        result = yield* race({
-          succeeded: take(buyCryptoSucceeded),
-          failed: take(buyCryptoFailed),
-          canceled: take(buyCryptoCanceled)
-        })
-      } else {
-        yield* put(
-          onrampOpened({
-            provider: USDCOnRampProvider.STRIPE,
-            purchaseInfo: {
-              desiredAmount: balanceNeededCents
-            }
-          })
-        )
-
-        result = yield* race({
-          succeeded: take(buyUSDCFlowSucceeded),
-          canceled: take(onrampCanceled),
-          failed: take(buyUSDCFlowFailed)
-        })
-      }
-      // Return early for failure or cancellation
-      if (result.canceled) {
-        yield* put(purchaseCanceled())
-        return
-      }
-      if (result.failed) {
-        yield* put(
-          purchaseContentFlowFailed({ error: result.failed.payload.error })
+      yield* call(purchaseUSDC, { amount: totalAmountDueCentsBN })
```

Comment:
Thoughts on having `purchaseUSDC` be named `doBuyUSDC`? Purchase/buy are getting used pretty interchangeably